src/main/java/htsjdk/samtools/cram/structure/Slice.java

```diff
@@ -75,16 +75,51 @@
 
     // for indexing purposes
 
+    public static final int UNINITIALIZED_INDEXING_PARAMETER = -1;
+
     // the Slice's offset in bytes from the beginning of its Container
     // equal to Container.landmarks[Slice.index] of its enclosing Container
-    public int offset = -1;
+    // BAI and CRAI
+    public int byteOffsetFromContainer = UNINITIALIZED_INDEXING_PARAMETER;
     // this Slice's Container's offset in bytes from the beginning of the stream
-    // equal to Container.offset of its enclosing Container
-    public long containerOffset = -1;
+    // equal to Container.byteOffset of its enclosing Container
+    // BAI and CRAI
+    public long containerByteOffset = UNINITIALIZED_INDEXING_PARAMETER;
     // this Slice's size in bytes
-    public int size = -1;
-    // this Slice's index within its Container
-    public int index = -1;
+    // CRAI only
+    public int byteSize = UNINITIALIZED_INDEXING_PARAMETER;
+    // this Slice's index number within its Container
+    // BAI only
+    public int index = UNINITIALIZED_INDEXING_PARAMETER;
+
+    void baiIndexInitializationCheck() {
```

Comment:
This looks like a good check, but it doesn't appear to be called from anywhere except within one test (`craiIndexInitializationCheck` below *is* called before indexing).

Reply:
I started looking at this a bit more. It looks like `populateSlicesAndIndexingParameters` is only called when we're creating an index for a CRAM file that we've read in from disk, but I'm unclear on how the slice index info gets populated in the case where we're creating the index on-the-fly, when writing a new CRAM file. Since the tests were passing, I assumed it was happening somehow. So, I naively tried adding a call to this method from within CRAMBAIIndexer.processContainer, and then ran as many of the indexing tests as I easily could. 6 of them failed. So this led to a few questions:
- Where should the call to `baiIndexInitializationCheck` method go ?
- Does the equivalent of `populateSlicesAndIndexingParameters` happen in the case where we're writing a cram out from SAMRecords ?
- If we aren't populating the slice index info when writing a new CRAM file on this branch, how/why is it that all of the index tests are still passing ?
- Is there a similar issue already on master that isn't being caught by tests ?